docs/source/examples/performance-training.rst

```diff
@@ -0,0 +1,402 @@
+.. _high-performance-training:
+
+High Performance Training
+=========================
+
+This guide covers best practices and examples for achieving high performance distributed training using SkyPilot.
+
+Distributed training basics
+----------------------------
+
+SkyPilot supports various distributed training frameworks and approaches:
+
+- `PyTorch Distributed Data Parallel (DDP) <https://github.com/skypilot-org/skypilot/tree/master/examples/spot/resnet_ddp>`_
+- `Horovod <https://github.com/skypilot-org/skypilot/tree/master/examples/horovod>`_
+- `DeepSpeed <https://github.com/skypilot-org/skypilot/tree/master/examples/deepspeed>`_
+- `Ray Train <https://github.com/skypilot-org/skypilot/tree/master/examples/ray_train>`_
+- `TensorFlow Distribution Strategies <https://github.com/skypilot-org/skypilot/tree/master/examples/tensorflow>`_
+
+The choice of framework depends on your specific needs, but all can be easily configured through SkyPilot's YAML specification.
+
+Best practices
+--------------
+
+High performance instances
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Choose high performance instances for optimal training performance. SkyPilot allows you to specify instance types with powerful GPUs and high-bandwidth networking:
+
+- Use the latest GPU accelerators (A100, H100, etc.) for faster training
+- Select instances with NVLink for improved multi-GPU communication
+- Consider instances with higher memory bandwidth for large models
+
+Example configuration:
+
+.. code-block:: yaml
+
+  resources:
+    accelerators: 
+      A100:1  
+      A100-80GB:1  
+      H100:1  
+
+Use best disk_tier
+~~~~~~~~~~~~~~~~~~~~~~~
+Fast storage is critical for data-intensive workloads in loading and storing data and model checkpoints.
+SkyPilot's disk tier options to select the fastest available storage with high-performance local SSDs to reduce I/O bottlenecks
+
+Example configuration:
+
+.. code-block:: yaml
+
+  resources:
+    disk_tier: best # using highest performance disk tier
+
+
+MOUNT_CACHED for checkpointing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MOUNT_CACHED provides high performance writing, making it ideal for model checkpoints, logs, and other outputs with fast local writes. Unlike MOUNT mode, it supports all write operations without limitations on file operations. It offers fast file access through a local VFS (Virtual File System) cache implemented by `rclone <https://rclone.org/>`__ that provides near-local disk performance. 
+
+Example configuration:
+
+.. code-block:: yaml
+
+    file_mounts:
+      /checkpoint:
+        name: my-checkpoint-bucket  
+        mode: MOUNT_CACHED
+
+.. note::
+   
+   In MOUNT_CACHED mode, files only begin uploading after they are closed by all processes. By default, SkyPilot uses sequential transfers to maintain file order, which can cause the cache to grow if files are written faster than they can be uploaded.
```

Comment:
Discussed:
- move the whole note to Buckets page
- readers may not have a mental model at this point yet to understand 'sequential transfers'